benchmark/custom_csv_formater.hpp

```diff
@@ -0,0 +1,182 @@
+#pragma once
+
+#include <benchmark/benchmark.h>
+#include <vector>
+namespace benchmark {
+
+class customCSVReporter : public BenchmarkReporter {
+public:
+  customCSVReporter() : printed_header_(false) {}
+  bool ReportContext(const Context &context) override;
+  void ReportRuns(const std::vector<Run> &reports) override;
+
+private:
+  std::string CsvEscape(const std::string &s) {
+    std::string tmp;
+    tmp.reserve(s.size() + 2);
+    for (char c : s) {
+      switch (c) {
+      case '"':
+        tmp += "\"\"";
+        break;
+      default:
+        tmp += c;
+        break;
+      }
+    }
+    return '"' + tmp + '"';
+  }
+
+  // Function to return an string for the calculated complexity
+  std::string GetBigOString(BigO complexity) {
```

Comment:
This is not a big deal, but I think the parameter (complexity) could be made const here, since it's not modified inside the function.

Reply:
done